api/v1alpha1/component_types.go

```diff
@@ -62,103 +56,27 @@ type ComponentSpec struct {
 	Secret string `json:"secret,omitempty"`
 
 	// Source describes the Component source
-	Source ComponentSource `json:"source,omitempty"`
+	Source *ComponentSource `json:"source,omitempty"`
 
 	// Compute Resources required by this component
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
 
 	// The number of replicas to deploy the component with
 	Replicas int `json:"replicas,omitempty"`
 
-	// The port to expose the component over
+	// The port to expose the component over. Referenced in generated service.yaml and route.yaml
 	TargetPort int `json:"targetPort,omitempty"`
 
-	// The route to expose the component with
+	// The route to expose the component with. Referenced in generated route.yaml
 	Route string `json:"route,omitempty"`
 
-	// An array of environment variables to add to the component
+	// An array of environment variables to add to the component.  EnvVar describes environment variables to use for the component
 	Env []corev1.EnvVar `json:"env,omitempty"`
 
-	// The container image to build or create the component from
-	ContainerImage string `json:"containerImage,omitempty"`
-
-	// Whether or not to bypass the generation of GitOps resources for the Component. Defaults to false.
-	SkipGitOpsResourceGeneration bool `json:"skipGitOpsResourceGeneration,omitempty"`
-}
-
-// ComponentStatus defines the observed state of Component
-type ComponentStatus struct {
-	// Condition about the Component CR
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// EnvironmentConfigEnvVar is an array of standard environment variables in addition to the component EnvVars.
+	// These will ONLY be added to the deployment patches overlays deployment.yaml
+	EnvironmentConfigEnvVar []corev1.EnvVar `json:"env"`
```

Comment:
what is this replacing? I am confused here

Also the json tag does not match the name, so the linter is throwing a warning here

Reply:
OK i see what this is trying to achieve. I think this will be confusing to leave it on the same level..

How about introducing an intermediate level here called Environment?

so component.environment.environmentconfig is semantically more easier to understand.. WDYT

Reply:
Using BaseEnvVar and OverlayEnvVar